### PR TITLE
Lvjav442 15

### PR DESCRIPTION
**Pull request by myzonchuk**

### CODE REVIEW

File: src/main/java/com/softserve/ita/java442/cityDonut/model/Category.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
+package com.softserve.ita.java442.cityDonut.model;
+
+import com.sun.istack.NotNull;
+
+import javax.persistence.*;
+import java.util.List;
+import java.util.Objects;
+
+@Entity
```
Comment (NazarRyvak) [inline src/main/java/com/softserve/ita/java442/cityDonut/model/Category.java:8]: add @Data @AllArgsConstructor @NoArgsConstructor

File: src/main/java/com/softserve/ita/java442/cityDonut/model/Category.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+@Table(name = "category")
+public class Category {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long id;
+    @NotNull
```
Comment (NazarRyvak) [inline src/main/java/com/softserve/ita/java442/cityDonut/model/Category.java]: remove 15 row and add @Column(name = "category", length = 35, nullable = false)

File: src/main/java/com/softserve/ita/java442/cityDonut/model/Category.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    @Column(name = "category", length = 35)
+    private String category;
+    @ManyToMany(mappedBy = "categories ")
+    private List<Project> projects;
+
+
```
Comment (NazarRyvak) [inline src/main/java/com/softserve/ita/java442/cityDonut/model/Category.java]: remove from 20 row to 66row... we use @Data from Lombok

File: src/main/java/com/softserve/ita/java442/cityDonut/model/Category.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import javax.persistence.*;
+import java.util.List;
+import java.util.Objects;
+
+@Entity
+@Table(name = "category")
+public class Category {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long id;
```
Comment (NazarRyvak) [inline src/main/java/com/softserve/ita/java442/cityDonut/model/Category.java:20]: add between your fields empty row... Easy to read

File: src/main/resources/application.properties [hunk 5]
```diff
@@ -1,4 +1,7 @@
-spring.jpa.hibernate.ddl-auto=
```
Comment (NazarRyvak) [inline src/main/resources/application.properties:1]: you need to remove application.properties from this pull request

File: src/main/java/com/softserve/ita/java442/cityDonut/model/Category.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+public class Category {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long id;
+
+    @Column(name = "category", length = 35, nullable = false)
+    private String category;
+
+    @ManyToMany(mappedBy = "categories")
```
Comment (NazarRyvak) [inline src/main/java/com/softserve/ita/java442/cityDonut/model/Category.java]: mappedBy = "category", fetch = FetchType.LAZY

**Review by NazarRyvak — APPROVED** [review]

**Review by martavoloshyn — APPROVED** [review]

**Pull request merged**